File: Solutions/Neetcode150/ArraysAndHashing/2-ValidAnagram/solution.ts
export function isAnagram(s: string, t: string): boolean {
  let sCharFreq = new Map<string, number>();
  let tCharFreq = new Map<string, number>();

  if (s.length !== t.length) return false;

  for (let i = 0; i < s.length; i++) {
    sCharFreq.set(s[i], (sCharFreq.get(s[i]) || 0) + 1);
    tCharFreq.set(t[i], (tCharFreq.get(t[i]) || 0) + 1);
  }

  for (const [sChar, sFreq] of sCharFreq) {
    if (tCharFreq.get(sChar) !== sFreq) return false;
  }

  return true;
}

File: Solutions/Neetcode150/ArraysAndHashing/2-ValidAnagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sFreq = {}
        tFreq = {}

        if len(s) != len(t):
            return False

        for i in range(len(s)):
            sFreq[s[i]] = sFreq.get(s[i], 0) + 1
            tFreq[t[i]] = tFreq.get(t[i], 0) + 1

        for char, freq in sFreq.items():
            if freq != tFreq.get(char):
                return False

        return True
